tests/Response/ResultsTest.php

```diff
@@ -0,0 +1,28 @@
+<?php declare(strict_types = 1);
+
+namespace ekinhbayar\GitAmpTests\Response;
+
+use Amp\Artax\Response;
+use PHPUnit\Framework\TestCase;
+use ekinhbayar\GitAmp\Events\Factory;
+use ekinhbayar\GitAmp\Response\Results;
+use ekinhbayar\GitAmp\Response\DecodingFailedException;
+
+class ResultsTest extends TestCase
+{
+    public function testAppendResponseThrowsOnInvalidJSON()
+    {
+        $response = $this->createMock(Response::class);
+
+        $response
+            ->expects($this->once())
+            ->method('getBody')
+            ->will($this->returnValue('[{"message"}]'))
+        ;
+
```

Comment:
This probably misses something to be actually useful and that I can't think of at this hour.